src/java/org/apache/cassandra/db/NativeClustering.java

```diff
@@ -45,7 +46,7 @@ public NativeClustering(NativeAllocator allocator, OpOrder.Group writeOp, Cluste
         int bitmapSize = ((count + 7) >>> 3);
 
         assert count < 64 << 10;
-        assert dataSize < 64 << 10;
+        assert dataSize <= FBUtilities.MAX_UNSIGNED_SHORT : String.format("Data size %d >= %d", dataSize, FBUtilities.MAX_UNSIGNED_SHORT + 1);
```

Comment:
Nice :)

Reply:
this shouldn't happen anymore as I added this check earlier in the pipeline... found some fun bugs with composite partitions where the values are > `Short.MAX_VALUE` and seems I had to be careful how we defend against it as composite columns allow things non-composite columns don't!  (you can have empty values when `PRIMARY KEY((a, b))` but not `PRIMARY KEY(a)`